extensions/guacamole-auth-jdbc/modules/guacamole-auth-jdbc-base/src/main/java/org/apache/guacamole/auth/jdbc/tunnel/RestrictedGuacamoleTunnelService.java

```diff
@@ -179,9 +187,27 @@ protected ModeledConnection acquire(RemoteAuthenticatedUser user,
 
             @Override
             public int compare(ModeledConnection a, ModeledConnection b) {
-
-                return getActiveConnections(a).size()
-                     - getActiveConnections(b).size();
+                
+                int weightA, weightB;
+                // Check if weight of a is non-null and retrieve it.
+                if (a.getConnectionWeight() != null && a.getConnectionWeight().intValue() > 0)
```

Comment:
Checking for `null`, using a default value, etc. seems the sort of logic that should be within `ModeledConnection`, rather than repeated here. That way you could simply call `a.getWeight()` or `a.getConnectionWeight()` and it will always just work?

Only the underlying model object (and the code applying attribute changes) need actually worry about such internal concerns.